File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/ledger/RevocGetRegDeltaRequestTest.java
package org.hyperledger.indy.sdk.ledger;

import org.hyperledger.indy.sdk.utils.PoolUtils;
import org.json.JSONObject;
import org.junit.Test;

import java.util.Date;

public class RevocGetRegDeltaRequestTest extends LedgerIntegrationTest {

	@Test
	public void testBuildGetRevocRegDeltaRequestWorks() throws Exception {
		JSONObject expectedResult = new JSONObject()
				.put("operation", new JSONObject()
						.put("type", "117")
						.put("revocRegDefId", revRegDefId)
						.put("to", 100)
				);

		String request = Ledger.buildGetRevocRegDeltaRequest(DID, revRegDefId, - 1, 100).get();

		assert (new JSONObject(request).toMap().entrySet()
				.containsAll(
						expectedResult.toMap().entrySet()));
	}

	@Test
	public void testRevocRegRequestsDeltaWorks() throws Exception {
		postEntities();

		int to = (int) (new Date().getTime()/1000) + 100;

		String getRevRegRequest = Ledger.buildGetRevocRegDeltaRequest(DID, revRegDefId, -1, to).get();
		String getRevReResponse = PoolUtils.ensurePreviousRequestApplied(pool, getRevRegRequest, response -> {
			JSONObject responseObject = new JSONObject(response);
			return !responseObject.getJSONObject("result").isNull("seqNo");
		});

		Ledger.parseGetRevocRegDeltaResponse(getRevReResponse).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/LedgerTests/RevocGetRegDeltaRequestTest.cs
using Hyperledger.Indy.LedgerApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Newtonsoft.Json.Linq;
using System;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.LedgerTests
{
    [TestClass]
    public class RevocGetRegDeltaRequestTest : LedgerIntegrationTestBase
    {
        [TestMethod]
        public async Task TestSubmitRequestWorks()
        {
            var expectedResult =
                "\"operation\":{" +
                        "\"type\":\"117\"," +
                        "\"revocRegDefId\":\"RevocRegID\"," +
                        "\"to\":100" +
                        "}";

            var request = await Ledger.BuildGetRevocRegDeltaRequestAsync(DID, "RevocRegID", -1, 100);

            Assert.IsTrue(request.Replace("\\s+", "").Contains(expectedResult.Replace("\\s+", "")));
        }

        [TestMethod]
        public async Task TestRevocRegRequestsDeltaWorks()
        {
            await PostEntitiesAsync();

            var timespan = (DateTime.UtcNow - new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc));
            var to = (long)timespan.TotalSeconds + 100;

            var getRevRegRequest = await Ledger.BuildGetRevocRegDeltaRequestAsync(DID, revRegDefId, -1, to);
            var getRevRegResponse = await PoolUtils.EnsurePreviousRequestAppliedAsync(pool, getRevRegRequest, response => {
                var responseObject = JObject.Parse(response);
                return responseObject["result"]["seqNo"] != null;
            });

            await Ledger.ParseGetRevocRegDeltaResponseAsync(getRevRegResponse);
        }
    }
}
